client/src/components/Picture/UploadPic.js

```diff
@@ -0,0 +1,211 @@
+import React, { useRef, useState } from "react";
+import { Modal } from "antd";
+import { CameraOutlined } from "@ant-design/icons";
+import axios from "axios";
+import BaseButton from "../Button/BaseButton";
+import ReactCrop from "react-image-crop";
+import styled from "styled-components";
+import { theme } from "../../constants/theme";
+
+const { colors } = theme;
+
+const CustomSubmitButton = styled(BaseButton)`
+  background: ${colors.royalBlue};
+  color: ${colors.white};
+`;
+
+const CustomCancelButton = styled(BaseButton)``;
+
+function isImageFile(file) {
+  return file && file["type"].split("/")[0] === "image";
+}
+
+const UploadPic = ({ cameraIconSize, user }) => {
+  const [modalVisible, setModalVisible] = useState(false);
+  const [photoURL, setPhotoURL] = useState();
+  const [uploadError, setUploadError] = useState();
+  const [image, setImage] = useState();
+  const [crop, setCrop] = useState({
+    aspect: 1 / 1,
+    unit: "px",
+    height: 250,
+    width: 250,
+  });
+  const imgUpload = useRef(null);
+
+  const imageLoaded = (image) => {
+    setImage(image);
+  };
+
+  const handleImage = (e) => {
+    setUploadError("");
+    const file = e.target.files[0];
+    if (file && file.size > 5001520) {
+      setUploadError("Please upload an image of size less than 5 MB.");
+    } else if (!isImageFile(file)) {
+      setUploadError(`Sorry, we only support image files.`);
+    } else {
+      const fileReader = new FileReader();
+      fileReader.onloadend = () => {
+        setPhotoURL(fileReader.result);
+      };
+      if (file) {
+        fileReader.readAsDataURL(file);
+      }
+    }
+    setModalVisible(true);
+  };
+
+  const savePhoto = async () => {
+    const canvas = document.createElement("canvas");
+    const scaleX = image.naturalWidth / image.width;
+    const scaleY = image.naturalHeight / image.height;
+    canvas.width = crop.width;
+    canvas.height = crop.height;
+    const ctx = canvas.getContext("2d");
+    ctx.drawImage(
+      image,
+      crop.x * scaleX,
+      crop.y * scaleY,
+      crop.width * scaleX,
+      crop.height * scaleY,
+      0,
+      0,
+      crop.width,
+      crop.height,
+    );
+    const reader = new FileReader();
+    canvas.toBlob((blob) => {
+      let uploadResponse;
+      let endPoint = `/api/users/current/avatar`;
+      let formData = new FormData();
```

Comment:
Minor: Use `const` for `formData` since it is not being reassigned to a different value.